### ISSUE TITLE
Show reset "x" during search

### ISSUE THREAD

**Issue by hoggatt** [m-wrzr/streamlit-searchbox#62]
## Current State

When a user is searching, he may click off of the searchbox to do something else. Especially if using `edit_after_submit="current/option/concat"`, the search result stays in the searchbox. It would be nice if the "x" reset option is still visible even when a search is happening.

## Request

It would be awesome to have the reset button visible even when an item is not selected (ie, just a search has been performed)
![image](https://github.com/user-attachments/assets/c1893dee-2f7a-4c07-8125-e1699e091dd0)

Ideally it is also visible even during a search:
![image](https://github.com/user-attachments/assets/2c10648d-e093-4690-9982-9720e46210e8)
But I understand that this may have some more downsides or be tougher to implement.

**Comment by hoggatt** [comment]
Sorry I'm not able to be more helpful on this one, I don't know TS/JS very well.

**Comment by m-wrzr** [comment]
Hey, I've made this part of `0.1.17`:

```
st_searchbox(
    search_function=search,
    style_overrides=StyleOverrides(
        clear={"clearable": "always"},
    ),
)
```

if you set it to `clearable` (other options are `never` / `after-submit`) this will show as long as you have something in the search box regardless of any selected options. hope that helps!

**Comment by hoggatt** [comment]
This is amazing! Thanks so much!

Closed by #61
